src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
1. It is better to avoid names with one letter, in this case you can use the destructuring
https://mate-academy.github.io/style-guides/javascript.html#51-use-object-destructuring-when-accessing-and-using-multiple-properties-of-an-object
2. The assignment says not to use forEach, but instead you can use reduce to find the sum
https://mate-academy.github.io/style-guides/javascript.html#11-iterators-and-generators